src/rdd/rdd.rs

```diff
@@ -838,6 +859,65 @@ pub trait Rdd: RddBase + 'static {
         self.intersection_with_num_partitions(other, self.number_of_splits())
     }
 
+    fn subtract<T>(&self, other: Arc<T>) -> SerArc<dyn Rdd<Item = Self::Item>>
+    where
+        Self: Clone,
+        Self::Item: Data + Eq + Hash,
+        T: Rdd<Item = Self::Item> + Sized,
+    {
+        self.subtract_with_num_partition(other, self.number_of_splits())
+    }
+
+    fn subtract_with_num_partition<T>(
+        &self,
+        other: Arc<T>,
+        num_splits: usize,
+    ) -> SerArc<dyn Rdd<Item = Self::Item>>
+    where
+        Self: Clone,
+        Self::Item: Data + Eq + Hash,
+        T: Rdd<Item = Self::Item> + Sized,
+    {
+        let other = other
+            .map(Box::new(Fn!(
+                |x: Self::Item| -> (Self::Item, Option<Self::Item>) { (x, None) }
+            )))
+            .clone();
+        let rdd = self
+            .map(Box::new(Fn!(|x| -> (Self::Item, Option<Self::Item>) {
+                (x, None)
+            })))
+            .cogroup(
+                other,
+                Box::new(HashPartitioner::<Self::Item>::new(num_splits))
+                    as Box<dyn Partitioner>,
+            )
+            .map(Box::new(Fn!(|(x, (v1, v2)): (
+                Self::Item,
+                (Vec::<Option<Self::Item>>, Vec::<Option<Self::Item>>)
+            )|
+             -> Option<Self::Item> {
+                if (v1.len() >= 1) ^ (v2.len() >= 1) {
+                    Some(x)
+                } else {
+                    None
+                }
+            })))
+            .map_partitions(Box::new(Fn!(|iter: Box<
+                dyn Iterator<Item = Option<Self::Item>>,
+            >|
+             -> Box<
```

Comment:
I maybe wrong, but it looks like rustfmt is not done. If so, please run cargo fmt.

Reply:
done